components/dialog/dialog-mixin.js

```diff
@@ -139,11 +140,16 @@ export const DialogMixin = superclass => class extends RtlMixin(superclass) {
 
 	_close(action) {
 		if (!this._state) return;
-		this._action = action;
+		this._action = action ?? abortAction;
 
 		clearDismissible(this._dismissibleId);
 		this._dismissibleId = null;
 
+		if (this._isCloseAborted()) {
+			this._dismissibleId = setDismissible(() => this._close(abortAction));
```

Comment:
What's the reasoning here behind clearing and setting a new dismissible instead of just preventing the original from being dismissed?

Reply:
As I understand, this is how init() method triggered by setDismissible() method works. 
https://github.com/BrightspaceUI/core/blob/bc58892cb5d4c32f82a4220c7ff638f9de0bbdd5/helpers/dismissible.js#L19
Each time when Esc is pressed the callback function is deleted from the stack.

Reply:
When the dialog is triggered to close by pressing Esc the callback function (_close) is deleted from the stack but if the close is aborted the dialog cannot be close by Esc the next time.